externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala

```diff
@@ -125,7 +126,12 @@ class CatalogShim_v2_4 extends SparkCatalogShim {
     databases.flatMap { db =>
       val identifiers = catalog.listTables(db, tablePattern, includeLocalTempViews = true)
       catalog.getTablesByName(identifiers).flatMap { t =>
-        t.schema.zipWithIndex.filter(f => columnPattern.matcher(f._1.name).matches())
+        var tableSchema: StructType = null;
+        tableSchema = t.schema
+        if (t.provider.getOrElse("") == "delta") {
+          tableSchema = spark.table(t.identifier.table).schema
+        }
+        tableSchema.zipWithIndex.filter(f => columnPattern.matcher(f._1.name).matches())
```

Comment:
How about

``` scala
val tableSchema = if (t.provider.getOrElse("") == "delta") {
  spark.table(t.identifier.table).schema
} else {
  t.schema
}
```

BTW, is 'delta' here case-sensitive?

Reply:
Yes, I changed it.  No, 'delta' is always lower case so I added equalsIgnoreCase.